text/0000-components.md

```diff
@@ -0,0 +1,261 @@
+read/query, Grid management and configuration (potentially including management
+and configuration of the Grid daemon itself), application key “escrow”, and any
+other functionality which is provided by Grid for applications.
+
+### State Delta Export
+
+The Grid state delta export component processes events from a Sawtooth instance
+and stores it into a reporting database.  The initial implementation will
+export to a PostgreSQL database instance which, in turn, provides the data for
+the REST API.
+
+The State Delta Export component will insert records into a database using
+a modified Type 2 Slowly Changing Dimensions (SCD) record schema.  Each record
+that has been exported from Sawtooth global state will include a "start" and
+"end" for when the value of the record is effective.  Unlike traditional SCD,
+the "start" and "end" values are block heights instead of timestamps.
+
+![State Delta Example Table](../images/grid_components_state_delta_1.svg)
+
+For example, suppose we are tracking the count of some items in global state.
+A materialized view of the data within PostgreSQL might look like:
+
+![Example Query Output](../images/grid_components_state_delta_2.svg)
+
+We can query this table to obtain the latest data with the following:
+
+```sql
+SELECT FROM counts_table name,count WHERE end = MAX_BLOCK;
+```
+
+![Example Query Output](../images/grid_components_state_delta_3.svg)
+
+And we could query the state as of block 3 with the following:
+
+```sql
+SELECT FROM counts_table name,count WHERE start <= 3 AND 3 < end;
+```
+
+![Example Query Output](../images/grid_components_state_delta_4.svg)
+
+Using this model, records may be joined across tables at a given block height.
+This provides consistent views for the data at any point in time.  For example,
+if a block is committed while paging records, the resulting count of records at
+the initial block height queried will remain the same.
+
+## Command Line Tool (grid)
+
+The Grid CLI (`grid`) provides an command-line interface for interacting with the
+Grid Daemon’s REST API, with the Grid CLI mirroring the functionality available
+in the REST API. For example, the Grid CLI will provide functionality to:
+
+- Submit transactions
+- Read/query schema data
+- Read/query Pike identity data
+- Read/query product data
+- Read/query track and trace data
+
+The CLI will also contain administrative commands, such as initializing
+a PostgreSQL instance with the necessary tables for State Delta Export.
+
+The Grid CLI should feel natural on a Linux system and adhere to Linux CLI
+norms and should be suitable for use when scripting. For example, when an error
+occurs, the CLI command should exit with a non-zero exit status so that it can
+be checked in a bash script. The command’s output should be both human friendly
+and easily parsable with common Linux command line tools such as awk and sed
+(when necessary, as different command-line options). 
+
```

Comment:
Any value in getting into things like CLI output format expectations, like `--output=yaml` for human-friendly views, `output=doc` for table view, and `output=json` for programmatic or logging?